src/common/Session.h

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************
+* Copyright (c) 2015 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the
+* Free Software Foundation, Inc.,
+* 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+***************************************************************************/
+
+#ifndef SDDM_SESSION_H
+#define SDDM_SESSION_H
+
+#include <QDir>
+#include <QSharedPointer>
+
+namespace SDDM {
+    class SessionModel;
+
+    class Session {
+    public:
+        enum Type {
+            UnknownSession = 0,
+            X11Session,
+            WaylandSession
+        };
+
+        explicit Session();
+        Session(Type type, const QString &name);
+
+        bool isValid() const;
+
+        Type type() const;
+
+        QString xdgSessionType() const;
+
+        QString name() const;
+
+        QDir directory() const;
+        QString fileName() const;
+
+        QString displayName() const;
+        QString comment() const;
+
+        QString exec() const;
+        QString tryExec() const;
+
+        QString desktopNames() const;
+
+        void setTo(Type type, const QString &name);
+
+        Session &operator=(const Session &other);
+
+    private:
+        bool m_valid;
+        Type m_type;
+        QDir m_dir;
+        QString m_name;
+        QString m_fileName;
+        QString m_displayName;
+        QString m_comment;
+        QString m_exec;
+        QString m_tryExec;
+        QString m_xdgSessionType;
+        QString m_desktopNames;
+
+        friend class SessionModel;
+    };
+
+    typedef QSharedPointer<Session> SessionPtr;
```

Comment:
this was used by SessionModel i just preserved, i think we can just do a new Session instead of a shared pointer.